.github/workflows/ci.yml

```
@@ -0,0 +1,90 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  fedora-libsoup2:
+    name: Test with libsoup 2
+    runs-on: ubuntu-latest
+    container: fedora:latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Check we are on Fedora
+      run: cat /etc/os-release
+    - name: Update
+      run: |
+        sudo dnf -y update
+    - name: Enable kalev/metrics copr
+      run: |
+        sudo dnf -y install 'dnf-command(copr)'
+        sudo dnf -y copr enable kalev/metrics
+    - name: Install dependencies
+      run: |
+        sudo dnf -y install \
+          gcc \
+          meson \
+          'pkgconfig(eosmetrics-0)' \
+          'pkgconfig(gio-unix-2.0)' \
+          'pkgconfig(glib-2.0)' \
+          'pkgconfig(gobject-2.0)' \
+          'pkgconfig(libsoup-2.4)' \
+          'pkgconfig(ostree-1)' \
+          'pkgconfig(polkit-gobject-1)' \
+          'pkgconfig(sqlite3)' \
+          'pkgconfig(systemd)' \
+          'pkgconfig(uuid)' \
+          python3-dbus \
+          python3-dbusmock \
+          ${NULL+}
+    - name: Meson setup
+      run: |
+        meson setup _build \
+          -Dsoup2=true \
```

Did you consider using a matrix strategy <https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs> rather than copy-pasting the job definition for soup2 and soup3 with just two minor changes?

I think the way you would write this is:

```yaml
jobs:
  fedora:
    strategy:
      matrix:
        soup2: ['true', 'false']
```

then on this line you can write:

```suggestion
          -Dsoup2=${{ matrix.soup2 }} \
```

and add something like the following two `steps:`:

```yaml
- name: Install soup2
  if: ${{ matrix.soup2 == 'true' }}
  run: |
    sudo dnf -y install 'pkgconfig(libsoup-2.4)'
- name: Install soup2
  if: ${{ matrix.soup2 == 'false' }}
  run: |
    sudo dnf -y install 'pkgconfig(libsoup-3.0)'
```